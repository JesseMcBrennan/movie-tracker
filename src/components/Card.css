.Card {
  width: 10rem;
  display: flex;
  flex-direction: column;
  margin: 2rem .2rem 0 1rem;
}



img {
  width: 10rem;
}

button {
  background: red;
  border: none;
  height: 1.5rem;
}