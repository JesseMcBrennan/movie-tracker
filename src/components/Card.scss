.Card {
  width: 15rem;
  display: flex;
  flex-direction: column;
  margin: 2rem 1rem 0 1rem;
  cursor: pointer;
  filter: brightness(70%);
  opacity: 0.95;
  min-height: 22rem;
  border: solid 1rem goldenrod;
  box-shadow: 0 0 80px #000000;
  border-radius: 24px;
  &:hover {
    filter: brightness(100%);
    opacity: 1;
  }
}

.Card article {
  display: none;
}

.details {
  width: 15rem;
  display: flex;
  flex-direction: column;
  margin: 2rem 1rem 0 1rem;
  cursor: pointer;
  filter: brightness(100%);
  opacity: 0.95;
  overflow: scroll;
  min-height: 22rem;
  border: solid 1rem goldenrod;
  border-radius: 24px;
}

.details article {
  background-color: black;
  color: #f7f7f7;
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 22.8rem;
  opacity: 0.8;
  position: absolute;
  transition: all 500ms ease;
  width: 15rem;
}

h2,
p {
  margin: 0;
  width: 13rem;
}

p {
  line-height: 1.4rem;
}

img {
  width: 15rem;
  min-height: 22rem;
  border: 2px solid black;
}

button {
  background: red;
  border: 2px solid black;
  box-shadow: 0 0 30px #000000;
  width: 15.36rem; 
  border-radius: 4px;
  height: 2.5rem;
  color: #f9f9f9;
  text-decoration: underline;
  font-size: 1rem;
  text-shadow: 2px 2px black;
  font-weight: 400;
  cursor: pointer;
}
