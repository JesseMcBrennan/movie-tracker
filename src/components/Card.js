import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addMovieToDatabase, deleteMovieFromDatabase } from '../utils/apiCalls';
import { toggleFavorite } from '../actions';
import PropTypes from 'prop-types'
import './Card.css';

class Card extends Component {
  constructor() {
    super();
    this.state = {
      toggle: false
    };
  }

  setFavoriteData = async movie => {
    const { user, favorites, handleToggle } = this.props;
    if (!favorites.includes(movie) && !favorites.includes(movie.id)) {
      handleToggle(movie);
      await addMovieToDatabase(user, movie);
    } else {
      handleToggle(movie);
      await deleteMovieFromDatabase(user, movie);
    }
  };

  render() {
    let { image, favorite, movie } = this.props;
    return (
      <div className={this.state.toggle ? 'details' : 'Card'}>
        <article onClick={() => this.setState({ toggle: !this.state.toggle })}>
          <h2>{movie.title}</h2>
          <p>{movie.overview}</p>
          <p>{movie.release}</p>
          <p>{movie.vote_average}</p>
        </article>
        <img
          onClick={() => this.setState({ toggle: !this.state.toggle })}
          src={image || favorite.image}
          alt="Murray Movie"
        />
        <button onClick={() => this.setFavoriteData(movie)}>
          ADD TO FAVORITES
        </button>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  handleToggle: movie => dispatch(toggleFavorite(movie))
});

export default connect(
  null,
  mapDispatchToProps
)(Card);

Card.propTypes = {
  image: PropTypes.string,
  favorite: PropTypes.object,
  movie: PropTypes.object,
  favorites: PropTypes.array,
  handleToggle: PropTypes.func
}
