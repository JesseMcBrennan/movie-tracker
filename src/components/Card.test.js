import React from 'react';
import { shallow } from 'enzyme';
import { mockMurrayData, mockMovie, mockUser } from '../utils/mockData/mockMurrayData'

import { Card } from './Card';

describe('Card', () => {
  
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Card movie={mockMovie} image={mockMovie.image}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})