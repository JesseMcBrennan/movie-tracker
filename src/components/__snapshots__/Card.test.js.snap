// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    image="https://image.tmdb.org/t/p/w500/4M6Lzl6HXvJAQg3jMTHdQ3EOnzX.jpg"
    movie={
        Object {
            "id": 164808,
            "image": "https://image.tmdb.org/t/p/w500/4M6Lzl6HXvJAQg3jMTHdQ3EOnzX.jpg",
            "overview": "Fear and Loathing on the Road to Hollywood, also known as Fear and Loathing in Gonzovision, is a documentary film producedby BBC in 1978 on the subject of Hunter S. Thompson, directed by Nigel Finch.  The road trip/film pairs Thompson with Finch's fellow Briton the illustrator Ralph Steadman. The party travel to Hollywood via Death Valley and Barstow from Las Vegas, scene of the pair's 1971 collaboration. It contains interviews with Thompson and Steadman, as well as some short excerpts from some of his work.",
            "release": "1978-11-02",
            "title": "Fear and Loathing on the Road to Hollywood",
            "vote_average": 8,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article
          onClick={[Function]}
>
          <h2>
                    Fear and Loathing on the Road to Hollywood
          </h2>
          <p>
                    Fear and Loathing on the Road to Hollywood, also known as Fear and Loathing in Gonzovision, is a documentary film producedby BBC in 1978 on the subject of Hunter S. Thompson, directed by Nigel Finch.  The road trip/film pairs Thompson with Finch's fellow Briton the illustrator Ralph Steadman. The party travel to Hollywood via Death Valley and Barstow from Las Vegas, scene of the pair's 1971 collaboration. It contains interviews with Thompson and Steadman, as well as some short excerpts from some of his work.
          </p>
          <p>
                    1978-11-02
          </p>
          <p>
                    8
          </p>
</article>,
        <img
          alt="Murray Movie"
          onClick={[Function]}
          src="https://image.tmdb.org/t/p/w500/4M6Lzl6HXvJAQg3jMTHdQ3EOnzX.jpg"
/>,
        <button
          onClick={[Function]}
>
          ADD TO FAVORITES
</button>,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Fear and Loathing on the Road to Hollywood
</h2>,
            <p>
              Fear and Loathing on the Road to Hollywood, also known as Fear and Loathing in Gonzovision, is a documentary film producedby BBC in 1978 on the subject of Hunter S. Thompson, directed by Nigel Finch.  The road trip/film pairs Thompson with Finch's fellow Briton the illustrator Ralph Steadman. The party travel to Hollywood via Death Valley and Barstow from Las Vegas, scene of the pair's 1971 collaboration. It contains interviews with Thompson and Steadman, as well as some short excerpts from some of his work.
</p>,
            <p>
              1978-11-02
</p>,
            <p>
              8
</p>,
          ],
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Fear and Loathing on the Road to Hollywood",
            },
            "ref": null,
            "rendered": "Fear and Loathing on the Road to Hollywood",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Fear and Loathing on the Road to Hollywood, also known as Fear and Loathing in Gonzovision, is a documentary film producedby BBC in 1978 on the subject of Hunter S. Thompson, directed by Nigel Finch.  The road trip/film pairs Thompson with Finch's fellow Briton the illustrator Ralph Steadman. The party travel to Hollywood via Death Valley and Barstow from Las Vegas, scene of the pair's 1971 collaboration. It contains interviews with Thompson and Steadman, as well as some short excerpts from some of his work.",
            },
            "ref": null,
            "rendered": "Fear and Loathing on the Road to Hollywood, also known as Fear and Loathing in Gonzovision, is a documentary film producedby BBC in 1978 on the subject of Hunter S. Thompson, directed by Nigel Finch.  The road trip/film pairs Thompson with Finch's fellow Briton the illustrator Ralph Steadman. The party travel to Hollywood via Death Valley and Barstow from Las Vegas, scene of the pair's 1971 collaboration. It contains interviews with Thompson and Steadman, as well as some short excerpts from some of his work.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1978-11-02",
            },
            "ref": null,
            "rendered": "1978-11-02",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 8,
            },
            "ref": null,
            "rendered": 8,
            "type": "p",
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Murray Movie",
          "onClick": [Function],
          "src": "https://image.tmdb.org/t/p/w500/4M6Lzl6HXvJAQg3jMTHdQ3EOnzX.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ADD TO FAVORITES",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "ADD TO FAVORITES",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article
            onClick={[Function]}
>
            <h2>
                        Fear and Loathing on the Road to Hollywood
            </h2>
            <p>
                        Fear and Loathing on the Road to Hollywood, also known as Fear and Loathing in Gonzovision, is a documentary film producedby BBC in 1978 on the subject of Hunter S. Thompson, directed by Nigel Finch.  The road trip/film pairs Thompson with Finch's fellow Briton the illustrator Ralph Steadman. The party travel to Hollywood via Death Valley and Barstow from Las Vegas, scene of the pair's 1971 collaboration. It contains interviews with Thompson and Steadman, as well as some short excerpts from some of his work.
            </p>
            <p>
                        1978-11-02
            </p>
            <p>
                        8
            </p>
</article>,
          <img
            alt="Murray Movie"
            onClick={[Function]}
            src="https://image.tmdb.org/t/p/w500/4M6Lzl6HXvJAQg3jMTHdQ3EOnzX.jpg"
/>,
          <button
            onClick={[Function]}
>
            ADD TO FAVORITES
</button>,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Fear and Loathing on the Road to Hollywood
</h2>,
              <p>
                Fear and Loathing on the Road to Hollywood, also known as Fear and Loathing in Gonzovision, is a documentary film producedby BBC in 1978 on the subject of Hunter S. Thompson, directed by Nigel Finch.  The road trip/film pairs Thompson with Finch's fellow Briton the illustrator Ralph Steadman. The party travel to Hollywood via Death Valley and Barstow from Las Vegas, scene of the pair's 1971 collaboration. It contains interviews with Thompson and Steadman, as well as some short excerpts from some of his work.
</p>,
              <p>
                1978-11-02
</p>,
              <p>
                8
</p>,
            ],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Fear and Loathing on the Road to Hollywood",
              },
              "ref": null,
              "rendered": "Fear and Loathing on the Road to Hollywood",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Fear and Loathing on the Road to Hollywood, also known as Fear and Loathing in Gonzovision, is a documentary film producedby BBC in 1978 on the subject of Hunter S. Thompson, directed by Nigel Finch.  The road trip/film pairs Thompson with Finch's fellow Briton the illustrator Ralph Steadman. The party travel to Hollywood via Death Valley and Barstow from Las Vegas, scene of the pair's 1971 collaboration. It contains interviews with Thompson and Steadman, as well as some short excerpts from some of his work.",
              },
              "ref": null,
              "rendered": "Fear and Loathing on the Road to Hollywood, also known as Fear and Loathing in Gonzovision, is a documentary film producedby BBC in 1978 on the subject of Hunter S. Thompson, directed by Nigel Finch.  The road trip/film pairs Thompson with Finch's fellow Briton the illustrator Ralph Steadman. The party travel to Hollywood via Death Valley and Barstow from Las Vegas, scene of the pair's 1971 collaboration. It contains interviews with Thompson and Steadman, as well as some short excerpts from some of his work.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1978-11-02",
              },
              "ref": null,
              "rendered": "1978-11-02",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 8,
              },
              "ref": null,
              "rendered": 8,
              "type": "p",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Murray Movie",
            "onClick": [Function],
            "src": "https://image.tmdb.org/t/p/w500/4M6Lzl6HXvJAQg3jMTHdQ3EOnzX.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ADD TO FAVORITES",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "ADD TO FAVORITES",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
